Firestore/core/test/firebase/firestore/remote/serializer_test.cc

```diff
@@ -1424,6 +1460,115 @@ TEST_F(SerializerTest, EncodesResumeTokens) {
   ExpectRoundTrip(model, proto);
 }
 
+TEST_F(SerializerTest, EncodesListenRequestLabels) {
+  core::Query q = Query("docs");
+
+  std::vector<
```

Comment:
Can you use `std::map<QueryPurpose, std::unordered_map<...>>`? (I'm using `map` just because it's shorter and efficiency doesn't matter here)